packages/clarity-js/src/data/metadata.ts

```diff
@@ -147,9 +150,23 @@ function num(string: string, base: number = 10): number {
 }
 
 function user(): string {
-  let id = getCookie(Constant.CookieKey);
-  // Splitting and looking up first part for forward compatibility, in case we wish to store additional information in a cookie
-  return id && id.length > 0 ? id.split(Constant.Pipe)[0] : shortid(); 
+  let cookie = getCookie(Constant.CookieKey);
+  if(cookie && cookie.length > 0) {
+    // Splitting and looking up first part for forward compatibility, in case we wish to store additional information in a cookie
+    let parts = cookie.split(Constant.Pipe);
+    let userId = cookie.split(Constant.Pipe)[0];
+    if (parts.length === 2) { return userId; } else {
+      // For backward compatibility introduced in v0.6.18; this else clause can be removed with future iterations
+      let deleted = `${Constant.Semicolon}${Constant.Expires}${(new Date(0)).toUTCString()}${Constant.Path}`;
+      // First, delete current user cookie which might be set on current sub-domain vs. root domain
+      document.cookie = `${Constant.CookieKey}=${deleted}`;
+      // Second, same thing for current session cookie so it can be re-written later with the root domain
+      document.cookie = `${Constant.SessionKey}=${deleted}`;
+      // Return the existing userId, so it can eventually be written with version info later
```

Comment:
>userId [](http://example.com/codeflow?start=29&length=6)

What if cookies are not enabled yet? Shold we not do this in that case?

Reply:
If cookies are not enabled then we wouldn't execute logic inside the if block and directly return a shortid. It will essentially be same as before - generating a new id.